sor/src/main/java/com/bazaarvoice/emodb/sor/core/DefaultMigratorTools.java

```diff
@@ -40,8 +49,21 @@ public void writeRows(String placement, Iterator<MigrationScanResult> results, R
     @Override
     public Iterator<MigrationScanResult> readRows(String placement, ScanRange scanRange) {
         checkNotNull(placement, "placement");
-        checkNotNull(scanRange, scanRange);
-        return _migratorReaderDao.readRows(placement, scanRange);
+        checkNotNull(scanRange, "scanRange");
+
+        // Since the range may wrap from high to low end of the token range we need to unwrap it
+        List<ScanRange> unwrappedRanges = scanRange.unwrapped();
+
+        Iterator<StashTokenRange> tokenRanges = Iterators.concat(
+                Iterators.transform(
+                        unwrappedRanges.iterator(),
+                        unwrappedRange -> _stashTableDao.getStashTokenRangesFromSnapshot(placement, placement, unwrappedRange.getFrom(), unwrappedRange.getTo())));
+
+        return Iterators.concat(
+                Iterators.transform(tokenRanges, tokenRange ->
+                        _migratorReaderDao.readRows(placement, ScanRange.create(tokenRange.getFrom(), tokenRange.getTo()))
+                )
```

Comment:
Was wondering if we can avoid duplicating this code using a common/helper method to use it here as well as on the stash end, but thats ok I think.

Reply:
I think it will be difficult because they using diferent underlying CQL/Astyanax operation to scan. Also, I've been trying to not modify stash code in the context of the migrator because we will likely be ripping it out of the codebase after the migration is complete.

Reply:
@bdevore17 agreed.